pkg/api/v1/conversion.go

```diff
@@ -480,10 +480,9 @@ func Convert_v1_PodTemplateSpec_To_api_PodTemplateSpec(in *PodTemplateSpec, out
 		in.Spec.InitContainers = values
 
 		// Call defaulters explicitly until annotations are removed
-		for i := range in.Spec.InitContainers {
-			c := &in.Spec.InitContainers[i]
-			SetDefaults_Container(c)
-		}
+		tmpPod := &Pod{Spec: PodSpec{InitContainers: values}}
+		SetObjectDefaults_Pod(tmpPod)
```

Comment:
yes, I think we need to continue to make it work until 1.8, although annotation is deprecated.